airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlQueryUtils.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+package io.airbyte.integrations.source.mssql;
+
+import static io.airbyte.cdk.integrations.source.relationaldb.RelationalDbQueryUtils.enquoteIdentifierList;
+import static io.airbyte.cdk.integrations.source.relationaldb.RelationalDbQueryUtils.getFullyQualifiedTableNameWithQuoting;
+import static io.airbyte.cdk.integrations.source.relationaldb.RelationalDbQueryUtils.getIdentifierWithQuoting;
+import static io.airbyte.integrations.source.mssql.MssqlSource.HIERARCHYID;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.microsoft.sqlserver.jdbc.SQLServerResultSetMetaData;
+import io.airbyte.cdk.db.jdbc.JdbcDatabase;
+import io.airbyte.cdk.db.jdbc.JdbcUtils;
+import io.airbyte.cdk.integrations.source.relationaldb.CursorInfo;
+import io.airbyte.cdk.integrations.source.relationaldb.models.CursorBasedStatus;
+import io.airbyte.cdk.integrations.source.relationaldb.models.InternalModels.StateType;
+import io.airbyte.cdk.integrations.source.relationaldb.state.StateManager;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteStream;
+import java.math.BigDecimal;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class to define constants related to querying mssql
+ */
+public class MssqlQueryUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MssqlQueryUtils.class);
+  private static final String MAX_OC_VALUE_QUERY =
```

Comment:
Ok if not handled in this PR : 

As we move towards a world where the initial snapshot code supports a resumable full refresh, should we do away with the "end" position queries? This was added in MySQL to prevent the case where a rapidly growing database would never have a full snapshot completed because records are added faster than we can read. In practice, I'm not sure this is an issue. 

So basically, what do you think of having repeated queries such as : 
`select * from table where pk >= previous_pk_val limit CHUNK_SIZE`

instead of 
`select * from table where pk >= previous_pk_val and pk < max_pk limit CHUNK_SIZE`

Reply:
Is this necessary? 
Since the first incremental sync is now queued immediately after initial completes, any record whose `pk > max_pk` is going to be included in that incremental sync.